File: packages/server-core/src/lib/movies/entities/person-entity.ts
import { Column, Entity, OneToMany } from 'typeorm';
import { EntityBase } from '../../common/entities/entity-base';
import { MovieRoleEntity } from './movie-role-entity';

@Entity()
export class PersonEntity extends EntityBase {
  @Column('string')
  firstName: string;

  @Column('string', { nullable: true })
  middleName: string;

  @Column('string')
  lastName: string;

  @OneToMany(() => MovieRoleEntity, (role) => role.person)
  roles: MovieRoleEntity[] = [];

  addRole(role: MovieRoleEntity) {
    this.roles = [...this.roles, role];
  }
}

File: src/House.Flix.Core/People/Entities/PersonEntity.cs
using House.Flix.Core.Common.Entities;
using House.Flix.Core.Movies.Entities;

namespace House.Flix.Core.People.Entities;

public class PersonEntity : IEntity
{
    public Guid Id { get; set; }

    public string FirstName { get; set; } = "";
    public string LastName { get; set; } = "";

    public List<MovieRoleEntity> MovieRoles { get; set; } = new();
}
